### DISCUSSION TITLE
How to log new metrics in on_validation_epoch_end

### DISCUSSION THREAD

**Discussion by Aceticia** [Lightning-AI/pytorch-lightning#8710] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hi all, I have a few validation dataloaders. Each of these logs a metric like "val_loss/dataset_idx_0". I would like to track the average of all validation loss for early stopping, but I can't seem to log new metrics in validation epoch end callbacks. How to log this new metric? Thank you so much.

**Comment by shivammehta25** [comment]
`self.log` is not working? If not then if you are logging to tensorboard you can directly use `self.logger.experiment.add_scalar()`.

**Reply by Aceticia** [reply]
Will early stopping call back have access to this logged metric?
